### PR TITLE
feat: initial sbt factory 

### PR DESCRIPTION
**Pull request by yashdesu**

### CODE REVIEW

File: contracts/SBTFactory.sol [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	function makeNewSBT(
+		address proxyAdmin,
+		bytes memory proxyCallData,
+		address minterUpdater,
+		address[] calldata minters,
+		bytes calldata identifier
+	) external onlyOwner returns (address) {
+		// Create the implementation.
+		address implementation = address(
+			new SBT{salt: keccak256(identifier)}()
```
Comment (KukretiShubham) [inline contracts/SBTFactory.sol:29]: is there a specific purpose of using salt here?

**Review by KukretiShubham — COMMENTED** [review]
looks good to me

**Review by aggre — APPROVED** [review]

File: contracts/SBTFactory.sol [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+				proxyCallData
+			)
+		);
+		emit SBTProxyCreated(identifier, proxy);
+
+		// Save the proxy created in mapping.
+		sbtProxyMapping[identifier] = address(proxy);
+
+		// Initialize the implementation.
+		SBT(implementation).initialize(minterUpdater, minters);
```
Comment (aggre) [inline contracts/SBTFactory.sol:47]: Initializing the implementation contract is not needed, I think.

Comment (yashdesu) [reply contracts/SBTFactory.sol:47]: Thanks, I've removed Initializing the implementation.

**Pull request merged**